### PR TITLE
Module tests and spec changes

### PR DESCRIPTION
**Pull request by sravand123**

### CODE REVIEW

File: src/networking/testSimulator/TestSimulator.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		if(args.length==0 || args[0]=="LAN") {
+				test = new LanTest();
+				communication = "LAN";
+		}
+		
+		else if(args[0]== "INTERNET") {
+			test = new InternetTest();
+			communication = "INTERNET";
+		}
+
```
Comment (convict-git) [inline src/networking/testSimulator/TestSimulator.java]: Else let the tester know that it expects some arguments.

File: src/networking/testSimulator/Message.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	
+	/**'
+	 * Method to check If two Message object are equal or not
+	 * @param m, a message object
+	 * @return boolean, returns if the param object is equal to current object or not
+	 */
+	public boolean equals(Message m) { 
+		  if(m.processingMessage.equals(this.processingMessage) && m.contentMessage.equals(this.contentMessage))return true;
+	      return false;
```
Comment (convict-git) [inline src/networking/testSimulator/Message.java]: Indentation.
The two conditions in if can be kept multiline.

File: src/networking/testSimulator/Client.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	
+	public Client(
+					  ICommunicator communicator,
+					  ClientInfo client1,
+					  ClientInfo client2,
+					  Message input,
+					  Message output,
+					  int numMessages,
+					  int msgLength,
+					  Stopper stopper
```
Comment (convict-git) [inline src/networking/testSimulator/Client.java]: Fix indentation to keep it consistent with other files.
If you don't see the difference on your local editor, check this on the GitHub file viewer, you will understand the difference.

File: src/networking/testSimulator/Client.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		
+		// Creating ProcessingManager 
+		DummyProcessor processor = new DummyProcessor(
+									    communicator,
+										srcClient,
+										destClient,
+										input,output,
+										numMessages,
+										msgLength,
+										stopper
```
Comment (convict-git) [inline src/networking/testSimulator/Client.java]: Fix indentation

File: src/networking/testSimulator/Client.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		DummyContent content = new DummyContent(
+										communicator,
+										srcClient,
+										destClient,
+										input,
+										output,
+										numMessages,
+										msgLength,
+										stopper
+									);
```
Comment (convict-git) [inline src/networking/testSimulator/Client.java]: Fix indentation

**Review by convict-git — CHANGES_REQUESTED** [review]
Also, every place in your module wherever you have `if`, `else`, `while`, `for` et cetera, should have `{ }` irrespective of how many statements it has inside it (be it one or more).

Eg. 
```diff
- if (condition) return true;
+ if (condition) {
+     return true;
+ }
```

File: src/networking/testSimulator/Client.java (hunk 5 above)
Comment (sravand123) [reply src/networking/testSimulator/Client.java]: @convict-git  ready to be merged

**Pull request merged**